### DISCUSSION TITLE
QInput rules from store

### DISCUSSION THREAD

**Discussion by MonkeyFGB** [quasarframework/quasar#14912] [General - Components / Directives / etc] [answered]
Hi
we want to define rules for a form, inside the store.
For example, if I have user's list section with add/edit form.
In this page/forms we use a dedicated store to communicate with API.

I can define rules inside store "users"

`const RULES = [val => !!val || 'Field is required'];`

and assign to Qinput through store

```
    <q-input
      ref="inputRef"
      filled
      v-model="model"
      label="Required Field"
      :rules="getRules"
    />
```

The problem is translate message with i18n component.

Is there a way to return the label from store 

`const RULES = [val => !!val || '**field_required**'];`

and with a callback function in Qinput call the $t component?

Thank you in advance

**Comment by arnonrdp** [comment, marked as answer]
I don't know how you are using your i18n but I think something like this might work:
```js
<q-input
    ref="inputRef"
    filled
    v-model="model"
    label="Required Field"
    :rules="[val => !!val || t('field_required')];"
/>
```

In addition, if the rule is working fine in the store, you can try something like this:
```js
<q-input
    ref="inputRef"
    filled
    v-model="model"
    label="Required Field"
    :rules="[getRules || t('field_required')];"
/>
```
